index.js

```diff
@@ -1,27 +1,33 @@
-module.exports = function (element, options) {
+module.exports = function (element, options, regex) {
   options = options || {}
   element.normalize()
+  regex = regex || null
 
   var tagName = options.tagName || 'span'
   var classPrefix = options.classPrefix != null ? options.classPrefix : 'char'
   var count = 1
 
   function inject (element) {
-    var parentNode = element.parentNode
     var string = element.nodeValue
-    var length = string.length
-    var i = -1
-    while (++i < length) {
+    var parentNode = element.parentNode
+    var letter_data = []
+    if (regex) {
+      letter_data = string.match(regex)
+    } else {
+      letter_data = string.split('')
+    }
+    var length = letter_data.length
+    for (var i = 0; i < length; i++) {
```

Comment:
Revert to the original way of writing a loop:

```js
var length = split.length
var i = -1
while (++i < length) {
  // ...
}
```

Reply:
Okay